fdbserver/ClusterController.actor.cpp

```diff
@@ -709,9 +709,24 @@ class ClusterControllerData {
 		for(int i = 0; i < proxies.size(); i++)
 			result.proxies.push_back(proxies[i].interf);
 
-		auto oldLogRouters = getWorkersForRoleInDatacenter( dcId, ProcessClass::LogRouter, req.maxOldLogRouters, req.configuration, id_used );
-		for(int i = 0; i < oldLogRouters.size(); i++) {
-			result.oldLogRouters.push_back(oldLogRouters[i].interf);
+		if(req.maxOldLogRouters > 0) {
+			auto oldLogRouters = tlogs;
```

Comment:
This seems to be assuming that `req.maxOldLogRouters` is less than or equal to `tlogs.size+1`. Either that, or we're just ignoring `maxOldLogRouters`.

Can we make that an `ASSERT` if that's the case?

Reply:
The code is already written so that it can recruit multiple log routers on the same worker, so the maxOlgLogRouters was a suggested maximum number of workers, but returning fewer is okay

Reply:
My concern was the opposite -- namely what if `maxOldLogRouters` is fewer than the number of logs. If such a thing were possible, this would result in there being more old log routers than the max.

Reply:
looking at the code, the list of workers does not determine the number of log routers, so in that case it would just ignore the additional interfaces